src/sortStudents.ts

```diff
@@ -1,16 +1,53 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageAge'
 }
 
 // create SortOrder type
-export type SortOrder;
+export enum SortOrder {
+  asc = 'asc',
+  desc = 'desc',
+}
 
+// eslint-disable-next-line no-multiple-empty-lines
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+// eslint-disable-next-line max-len
+export function sortStudents(students: Student, sortBy: SortType, order: SortOrder): [] {
+  if (order === SortOrder.asc) {
+    switch (sortBy) {
+      case SortType.Name || SortType.Surname:
+        return students.slice().sort((x, y) =>
+          x.sortBy.localeCompare(y.sortBy));
+      case SortType.Age || SortType.Married:
+        return students.slice().sort((x, y) => x.sortBy - y.sortBy);
+      default:
+        return students.slice().sort((x, y) =>
+          x.sortBy.reduce((sum, t) => sum + t, 0)
+          - y.sortBy.reduce((sum, t) => sum + t, 0));
+    }
+  } else {
+    switch (sortBy) {
+      case SortType.Name || SortType.Surname:
+        return students.slice().sort((x, y) =>
+          y.sortBy.localeCompare(x.sortBy));
+      case SortType.Age || SortType.Married:
+        return students.slice().sort((x, y) => y.sortBy - x.sortBy);
+      default:
+        return students.slice().sort((x, y) =>
```

Comment:
```suggestion
        return [...students].sort((x, y) =>
```
better make copy in such way

Reply:
and `student1`, `student2` instead of x and y